src/osd/PrimaryLogPG.cc

```diff
@@ -6801,7 +6801,11 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	if (result < 0)
 	  break;
 
-	ceph_assert(op.extent.length);
+	if (!op.extent.length) {
+	  result = -EINVAL;
```

Comment:
Have you considered making it a no-op instead?  IIRC a zero-length write is a no-op (but it does create an object as a side effect).  A zero-length zero could be a complete no-op, just like if the object isn't there.

Reply:
From a high-level perspective, it would be nice if write and zero behaved consistently for non-existent/whiteout objects but it looks like the consistent behaviour was reverted in commit 68863bb45324 ("osd: make ZERO on non-existent object a no-op") on purpose.

Reply:
> Have you considered making it a no-op instead? IIRC a zero-length write is a no-op (but it does create an object as a side effect). A zero-length zero could be a complete no-op, just like if the object isn't there.

No, I hadn't really considered it, but I'm fine with doing that instead if that's what you guys would prefer. Let me know.